kore/src/Kore/Step/Simplification/And.hs

```diff
@@ -176,19 +189,50 @@ makeEvaluateNonBool
         initialConditions = firstCondition <> secondCondition
         merged = Conditional.andCondition terms initialConditions
     normalized <- Substitution.normalize merged
-    return (applyAndIdempotence <$> normalized)
-        { predicate = applyAndIdempotence <$> Conditional.predicate normalized }
-
-applyAndIdempotence
+    return
+        (applyAndIdempotenceAndFindContradictions <$> normalized)
```

Comment:
So, I guess that normalized is a `Pattern`, and `applyAndIdempotenceAndFindContradictions` acts on its term. If so, I think that it would be more readable to update the `term` between curly braces below, as you do with the `predicate`, since you would use the same syntax for updating both fields.

I may be biased though, I never liked the functor instance for `Conditional` that much, so do whatever makes sense.

Reply:
The result would be:
```
    return
        normalized
            { term =
                applyAndIdempotenceAndFindContradictions
                    (Conditional.term normalized)
            , predicate =
                applyAndIdempotenceAndFindContradictions
                    <$> Conditional.predicate normalized
            }
```

I think I like this version better, thanks.